src/olympia/devhub/helpers.py

```diff
@@ -115,7 +115,17 @@ def docs_breadcrumbs(context, items=None):
 @jinja2.contextfunction
 def add_file_modal(context, title, action, action_label, modal_type='file'):
     addon = context['addon']
-    upload_url = reverse('devhub.upload_for_addon', args=[addon.slug])
+    version = context.get('version',
+                          addon.find_latest_version_including_rejected())
+    if version:
+        channel_id = version.channel
+    else:
+        # short term fix - this function won't be used after new file upload.
+        channel_id = (amo.RELEASE_CHANNEL_LISTED if addon.is_listed else
+                      amo.RELEASE_CHANNEL_UNLISTED)
+    channel = amo.CHANNEL_CHOICES_API[channel_id]
```

Comment:
It's just that `{something}_CHOICES_API` indicates that it's going to be used for the API - like we do for the status, for instance and as such may change if the API needs to change... which might be an issue if we have those values in some non-api URLs.

IHMO, if you're not dealing with an API, you should not use `{something}_CHOICES_API` at all. You're not gaining much by using this lookup dict anyway since the url names are hardcoded as `listed` and `unlisted` in `urls.py`